pbjs-twirp/twirp.ts

```diff
@@ -35,14 +35,24 @@ const getTwirpError = (err: AxiosError): TwirpError => {
     return twirpError;
 };
 
-export const createTwirpAdapter = (axios: AxiosInstance, methodLookup: (fn: any) => string): RPCImpl => {
+export interface adapterOptions {
+    methodLookup: (fn: any) => string,
+    headers: () => any
+};
+
+export const createTwirpAdapter = (axios: AxiosInstance, options: adapterOptions): RPCImpl => {
```

Comment:
createTwirpAdapter takes an AxiosInstance.  If you need to add custom headers, or modify the request in any way, the right place to do it is during the configuration of axios. For static headers you can just pass the object, and for dynamic headers there is transformRequest.

See: https://github.com/axios/axios#request-config

If that's the only functionality needed then this PR can likely be declined.

Reply:
Ok, that makes sense, but in the generated pbjs bindings you don't allow to customize the axios instance, it is created inside the create client func. Or is there some global static way to configure axios so all insances get the config? I think that would not suffice as it would then affect all other potential axios usage in the app.

```js
export const createService = (baseURL: string): somenamespace.Service => {
    const axios = Axios.create({
        baseURL: baseURL + MyServicePathPrefix,
        headers: {
          Accept: 'application/protobuf'
        }
    });

    return somenamespace.Service.create(createTwirpAdapter(axios, getServiceMethodName));
};
```

How exatly do I customize the axios with this type of generated binding? Should the library take in axios options in?

```js
export const createService = (baseURL: string, axiosOptions: any): somenamespace.Service => {
    const axios = Axios.create(Object.assign({
        baseURL: baseURL + MyServicePathPrefix,
        headers: {
          Accept: 'application/protobuf'
        }
    }, axiosOptions));
...
```

Reply:
The "createService" method is about 4 lines of code, and is just intended to be a convenience for common use cases.  I would just copy/paste and create your own method for your project to achieve this level of customization.

Between the createTwirpAdapter function, and the service constructor generated by protobuf-js, these are composable pieces that are easy to put together.

Reply:
Thanks for your answers.